xwiki-authenticator-trusted-api/src/main/java/org/xwiki/contrib/authentication/internal/DefaultTrustedAuthenticationConfiguration.java

```diff
@@ -231,4 +238,25 @@ private String stripContextPathFromURL(URL url)
         return StringUtils.removeStart(url.toExternalForm(),
             url.getProtocol() + "://" + url.getAuthority() + context.getWiki().getWebAppPath(context));
     }
+
+    @Override
+    public Collection<DynamicRoleConfiguration> getDynamicRoleConfigurations()
+    {
+        if (dynamicRoleConfigurations == null) {
+            String[] configurationNames = getCustomProperty(DYNAMIC_ROLE_CONFIGURATIONS_PROPERTY, "").split("\\|");
+            dynamicRoleConfigurations = new ArrayList<DynamicRoleConfiguration>(configurationNames.length);
+            for (String name : configurationNames) {
+                String prefix = DYNAMIC_ROLE_CONFIGURATION_PREFIX + name + ".";
+                DynamicRoleConfiguration conf = new DefaultDynamicRoleConfiguration(this, prefix, name);
+                if (conf.getGroupPrefix().isEmpty() && conf.getGroupSuffix().isEmpty()) {
+                    // Allowing configurations without group prefix or suffix would be dangerous
+                    // as it would match any group the user is in
+                    logger.error("Dynamic role configuration [{}] doesn't specify any group prefix or suffix.");
```

Comment:
Error seems a little high here, or if this is that bad it could deserve an exception?

Reply:
It is bad, because the system could allow access to users that should not because of a bad configuration / remove the user in unrelated groups like XWikiAllGroup.

It doesn't raise an exception, but this code returns false leading to the autenticate() method to deny access.

Reply:
I made the error message more explicit.